lib/private/docker.ts

```diff
@@ -55,14 +56,17 @@ export interface DockerCacheOption {
 export class Docker {
   private configDir: string | undefined = undefined;
 
-  constructor(private readonly logger?: Logger) {}
+  constructor(private readonly shellEventPublisher: ShellEventPublisher) {}
 
   /**
    * Whether an image with the given tag exists
    */
   public async exists(tag: string) {
     try {
-      await this.execute(['inspect', tag], { quiet: true });
+      await this.execute(['inspect', tag], {
+        subprocessOutputDestination: 'ignore',
+        shellEventPublisher: this.shellEventPublisher,
```

Comment:
`this.execute` probably already has access to `this.shellEventPublisher` ?

Reply:
Yeah it's a little silly, I fixed this by marking it as an optional prop of the shell options and then always passing it in the execute call just so we're not irrelevantly passing this everywhere

Reply:
What I mean is: the original implementation already did the right thing for `logger`: it always passed `shell({ logger: this.logger })`.

This seems like it's making the API worse. Why are we doing that?

Reply:
The comment 5 days ago is a bit murky to me what you mean but I'm taking it to mean "Yeah we don't need `eventEmitter` (or the like) and `subprocessOutputDestination` to be passed for every function call, which is absolutely true. Removed that